packages/ui/src/common/components/SmallFileUpload/SmallFileUpload.tsx

```diff
@@ -8,29 +8,54 @@ import { Row } from '@/common/components/Modal'
 import { ColumnGapBlock, RowGapBlock } from '@/common/components/page/PageContent'
 import { TextMedium } from '@/common/components/typography'
 import { BorderRad } from '@/common/constants'
+import { resizeImageFile } from '@/common/helpers'
 import { enhancedGetErrorMessage, enhancedHasError } from '@/common/utils/validation'
 import { Avatar } from '@/memberships/components/Avatar'
 import { SUPPORTED_IMAGES } from '@/memberships/model/validation'
 
 interface SmallFileUploadProps {
   name: string
-  onUpload: (event: ChangeEvent<HTMLInputElement>) => void
+  onUpload: (event: ChangeEvent<HTMLInputElement>, resizedImage?: Blob | null | File) => void
   initialPreview?: string
 }
 
 export const SmallFileUpload = ({ onUpload, name, initialPreview }: SmallFileUploadProps) => {
-  const [localValue, setLocalValue] = useState<File | null>()
+  const [localValue, setLocalValue] = useState<{ blob: Blob | null; file: File | null }>({
+    blob: null,
+    file: null,
+  })
   const { formState } = useFormContext()
   const [avatarPreview, setAvatarPreview] = useState<string>(initialPreview ?? '')
-
   useEffect(() => {
-    if (localValue && SUPPORTED_IMAGES?.includes(localValue.type)) {
-      const objectUrl = URL.createObjectURL(localValue)
+    if (localValue.blob && SUPPORTED_IMAGES?.includes(localValue.file?.type ?? '')) {
+      const objectUrl = URL.createObjectURL(localValue.blob)
       setAvatarPreview(objectUrl)
-      return () => URL.revokeObjectURL(objectUrl)
+
+      return () => {
+        objectUrl && URL.revokeObjectURL(objectUrl)
+      }
     }
   }, [localValue])
 
+  const onChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const file = event.target.files?.item(0) ?? null
+
+      if (file) {
+        try {
+          resizeImageFile(file, 192, 192, 'image/webp').then((blob) => {
+            setLocalValue({ blob, file })
+            onUpload(event, blob)
+          })
+        } catch (e) {
+          setLocalValue({ blob: null, file })
+          onUpload(event, file)
+        }
+      }
```

Comment:
Could this fix the validation ?
```suggestion
      if (file && SUPPORTED_IMAGES?.includes(file.type)) {
        try {
          resizeImageFile(file, 192, 192, 'image/webp').then((blob) => {
            setLocalValue({ blob, file })
            return onUpload(event, blob)
          })
        } catch (e) {
        }
      }
      setLocalValue({ blob: null, file })
      onUpload(event, file)
```